### PR TITLE
Frontend [new Draft]

### PR DESCRIPTION
**Pull request by nemi-notrace**
This is a new draft including radix-ui

### CODE REVIEW

File: packages/repco-frontend/app/lib/helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export function setStorage(name: string, value: string) {
+  console.log('SET')
```
Comment (dryptomim) [inline packages/repco-frontend/app/lib/helpers.tsx:2]: remove debug logs

File: packages/repco-frontend/app/lib/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  const values = localStorage.getItem(name)
+  const arr = values ? JSON.parse(values) : []
+  arr.push(value)
+  localStorage.setItem(name, JSON.stringify(arr))
+}
+
+export function getStorage(name: string) {
+  console.log('get')
```
Comment (dryptomim) [inline packages/repco-frontend/app/lib/helpers.tsx:10]: remove debug logs

File: packages/repco-frontend/app/routes/$playlists/add.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const formData = await request.formData()
+  const data = formData.get('add-item') || ''
+
+  const add = formData.get('add-to-playlist') || ''
+  return json({ uid: data, playlistAndUid: add })
+}
+
+export default function Playlists() {
+  const { uid, playlistAndUid } = useActionData()
+  console.log(uid, playlistAndUid)
```
Comment (dryptomim) [inline packages/repco-frontend/app/routes/$playlists/add.tsx:23]: remove debug logs

File: packages/repco-frontend/app/routes/$playlists/add.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }, [uid, playlistAndUid])
+
+  useEffect(() => {
+    if (playlistAndUid) {
+      const newPlaylist = [playlistAndUid.split(',')]
+      setStorage(newPlaylist[0][0], newPlaylist[0][1])
+      setPlaylists(getStorage('playlists'))
+      setadded(true)
+    }
+  }, [playlistAndUid, uid])
```
Comment (dryptomim) [inline packages/repco-frontend/app/routes/$playlists/add.tsx:37]: i think you can simplify this and dont have to use two useEffect hooks

File: packages/repco-frontend/app/routes/$playlists/playlist/$id.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import { getStorage } from '~/lib/helpers'
+
+export const loader: LoaderFunction = ({ params }) => {
+  const playlist = params
+  return playlist
+}
+
+export default function Playlist() {
+  const data = useLoaderData()
+  console.log(data)
```
Comment (dryptomim) [inline packages/repco-frontend/app/routes/$playlists/playlist/$id.tsx:13]: remove debug logs

File: packages/repco-frontend/app/routes/$playlists/playlist/$id.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  const [playlists, setPlaylists] = useState([])
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setPlaylists(getStorage(data.id))
+    }
+  }, [data])
+
+  console.log(data)
```
Comment (dryptomim) [inline packages/repco-frontend/app/routes/$playlists/playlist/$id.tsx:22]: remove debug logs

File: packages/repco-frontend/app/routes/playlists.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import type { ActionFunction, LoaderFunction } from '@remix-run/node'
```
Comment (dryptomim) [inline packages/repco-frontend/app/routes/playlists.tsx:1]: move some parts of this page to /$playlists/index and just leave the shared parts here and use the outlet-component

**Review by dryptomim — CHANGES_REQUESTED** [review]

**Comment by Frando** [comment]
Closing in favor of #38

**Pull request closed without merging**